src/transaction.cpp

```diff
@@ -943,7 +941,26 @@ bool wsrep::transaction::bf_abort(
 
     if (ret)
     {
+        if (is_streaming() &&
+            client_state_.mode() == wsrep::client_state::m_high_priority)
+        {
+            lock.unlock();
+            if (!client_state_.server_state().stop_streaming_applier(
```

Comment:
Doesn't this leak `wsrep::hight_priority_service` if the `stop_streaming_applier()` returns non-null? Shouldn't the returned instance be released?

Reply:
It doesn't leak, because the streaming applier is going to be released by rollbacker.
Notice that the same was happening prior to this patch.
Here I pretty much just moved the call stop_streaming_applier() to happen earlier, because if the streaming applier has already been taken by rollback fragment, then we simply bail out from bf_abort(), without making any changes to the transaction's state.

Reply:
I should probably amend the comment there to clarify this